tawazi/_dag/dag.py

```diff
@@ -417,25 +416,107 @@ def _assign_compound_priority(self) -> None:
             # assign the new leaf nodes
             leaf_ids = graph_ids.leaf_nodes
 
-    def draw(self, k: float = 0.8, display: bool = True, t: int = 3) -> None:
+    def draw(
+        self, show_nodes_names: bool = False, fig_name: str = "", layout_algorithm: str = "tree"
```

Comment:
```suggestion
        self, fig_name: str, show_nodes_names: bool = True, layout_algorithm: str = "tree"
```

1. we should show nodes names by default
2. lete's force the user to provide a figure name... This seems a bit simpler